src/components/mobile/barcodeScanner.tsx

```diff
@@ -53,10 +89,16 @@ export class BarcodeScanner extends React.PureComponent<
           getHeaders(token)
         )
         .then((res: any) => {
-          alert(JSON.stringify(res));
+          const asset_data = res.data.asset_data;
+          this.setState({
+            token: token,
```

Comment:
I addressed this in my overall PR comments but I want to clarify a little more here. For some reason, since this component uses the `useRef()` hook, `MapStatetoProps()` and `connect()` redux functions do not seem to be linking `token` to `this.props.token`. Because of this, I had to use the redux `useSelector()` hook, which is basically the functional programming equivalent of `MapStateToProps()`. I am able to store it in the `BarcodeScanner` state but can't find a way to save it in props. This seems to propagate to other components rendered from this page even if they do not have any  hooks, so I just decided to use `this.state.token` for all of our token needs and pass it in to components on the same page as a prop if needed.